app/models/set_member_subject.rb

```diff
@@ -53,7 +54,7 @@ def self.unseen_for_user_by_workflow(user, workflow)
   end
 
   def retire_workflow(workflow)
-    count = subject_workflow_counts.find_or_create_by!(workflow_id: workflow.id)
+    count = SubjectWorkflowCount.find_or_create_by!(workflow_id: workflow.id, subject_id: subject_id)
     count.retire!
```

Comment:
I'm opting not to touch queuing in this PR given how many fires we've been fighting around it, let me know if you think we should create an issue and follow up in another PR.